src/odemis/driver/tfsbc.py

```diff
@@ -0,0 +1,130 @@
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+
+from odemis import model
+import logging
+import serial
+from pymodbus.client.sync import ModbusSerialClient
+import numpy
+import serial.tools.list_ports
+
+# Parameters for connection
+BAUDRATE = 230400
+TIMEOUT = 1
+BYTESIZE = 8
+PARITY = serial.PARITY_NONE
+STOPBITS = 1
+
+# Modbus level addresses
+SLAVE_UNIT = 2
+
+# Modbus registers
+BEAMDEFL_LX = 0  # lower x beam deflection element control
+BEAMDEFL_LY = 1  # lower y beam deflection element control
+BEAMDEFL_UX = 2  # upper x beam deflection element control
+BEAMDEFL_UY = 3  # upper y beam deflection element control
+
+RANGE_AMPS = [-42.2e3, 42.2e3]
+
+class BeamController(model.HwComponent):
+    """
+    Driver for the Thermofischer beam controller.
+    This class provides the .shift VA containing a tuple of two floats which describe
+    the x and y beam offset in m.
+    """
+
+    def __init__(self, name, role, amps_per_m, zero_offset_amps,
+                 rs485_serial_num, **kwargs):
+        """
+        :param port (str): port (e.g. /dev/ttyUSB0)
+        :param amps_per_m (float): conversion factor amplitude/meter
+        :param zero_offset_amps (float): amplitude for which beam deflection is 0
+        :param rs485_serial_num (str): serial number of RS485 adapter
+        """
+        # .hwVersion, .swVersion not available
+        model.HwComponent.__init__(self, name, role, **kwargs)
+
+        self._amps_per_m = amps_per_m
+        self._zero_offset_amps = zero_offset_amps
+
+        # Find port by RS485 adapter serial number
+        for p in serial.tools.list_ports.comports():
+            if rs485_serial_num in p.hwid:
+                # Found it!
+                port = p
+                break
+        else:
+            raise IOError("RS484 adapter with serial number %s not found!" % rs485_serial_num)
+
+        # Modbus connection
+        self._modbus = ModbusSerialClient(method='rtu', port=port,
+                                          baudrate=BAUDRATE, timeout=TIMEOUT,
+                                          stopbits=STOPBITS, parity=PARITY,
+                                          bytesize=BYTESIZE)
+
+        # Calculate range
+        range_amps = numpy.subtract(RANGE_AMPS, zero_offset_amps)
+        range_m = numpy.divide(range_amps, amps_per_m)
+
+        # Default value: zero offset (in m)
+        default_shift = tuple(numpy.divide(zero_offset_amps, amps_per_m))
+        self.beamShift = model.TupleContinuous(default_shift, range_m,
```

Comment:
I'm the one who had suggested just naming it "shift". There are two (linked) reasons:
* That's the name we already use in the phenom driver (and it's nice that all VAs with the same behaviour have the same name)
* It's on a component a role like "ebeam-control", so it should be relatively obvious that we are talking for *beam* shift.